<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  <title>Stay hungry, stay foolish.</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="CLDIAO&apos;s blog">
<meta property="og:type" content="website">
<meta property="og:title" content="Stay hungry, stay foolish.">
<meta property="og:url" content="https://cldiao.github.io/index.html">
<meta property="og:site_name" content="Stay hungry, stay foolish.">
<meta property="og:description" content="CLDIAO&apos;s blog">
<meta property="og:locale" content="en">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Stay hungry, stay foolish.">
<meta name="twitter:description" content="CLDIAO&apos;s blog">
  
    <link rel="alternate" href="/atom.xml" title="Stay hungry, stay foolish." type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    
  
  <link rel="stylesheet" href="/css/style.css">
  

</head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    
    <div id="header-inner" class="inner">
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="https://cldiao.github.io"></form>
      </div>
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      
    </div>
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">Stay hungry, stay foolish.</a>
      </h1>
      
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main">
  
    <article id="post-Mac上安装一些应用" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/Mac上安装一些应用/" class="article-date">
  <time datetime="2018-09-01T06:51:45.000Z" itemprop="datePublished">Sep 1, 2018</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/Mac上安装一些应用/">Mac上安装一些应用</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <!-- Table of Contents -->
        
        <p><a href="https://www.vmware.com/go/getfusion" target="_blank" rel="noopener">Vmware fusion 10 download</a></p>
<p>激活码：FG3TU-DDX1M-084CY-MFYQX-QC0RD</p>
<p><a href="http://msdn.itellyou.cn/" target="_blank" rel="noopener">windows系统下载</a></p>
<p><a href="https://pan.baidu.com/s/1jxzGpv9NZtsvxeyGq3LmTA" target="_blank" rel="noopener">Windows系统绿化</a>  密码：6a5y (windows 中office也可使用此工具)</p>
<p><a href="https://www.adobe.com/cn/downloads.html?promoid=RL89NGY7&amp;mv=other" target="_blank" rel="noopener">Adobe下载</a></p>
<p><a href="https://pan.baidu.com/s/1LrTTR6D4zZ7o3WVXSppcDQ" target="_blank" rel="noopener">PS绿化工具</a>  密码：zv8c</p>
<p><a href="http://msdn.itellyou.cn/" target="_blank" rel="noopener">office2016下载</a></p>
<p><a href="https://pan.baidu.com/s/1LFRWxQiI5V9F6FJBf27gHw" target="_blank" rel="noopener">office 2016绿化工具</a>  密码：otve</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://cldiao.github.io/Mac上安装一些应用/" data-id="cjlkud5300000vomvxc03btj9" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Mac-office-Vmware-fusion-Ps/">Mac, office, Vmware fusion, Ps</a></li></ul>

    </footer>
  </div>
  
</article>
 


  
    <article id="post-RNA-Seq analysis in R" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/RNA-Seq analysis in R/" class="article-date">
  <time datetime="2018-07-28T16:28:44.000Z" itemprop="datePublished">Jul 29, 2018</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/RNA-Seq analysis in R/">EDITING</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <!-- Table of Contents -->
        
        <h1 id="RNA-Seq-analysis-in-R"><a href="#RNA-Seq-analysis-in-R" class="headerlink" title="RNA-Seq analysis in R"></a>RNA-Seq analysis in R</h1><h2 id="Some-preparation"><a href="#Some-preparation" class="headerlink" title="Some preparation:"></a>Some preparation:</h2><h3 id="Packages-used"><a href="#Packages-used" class="headerlink" title="Packages used:"></a>Packages used:</h3><p>limma</p>
<p>edgeR</p>
<p>gplots</p>
<p>org.Mm.eg.db</p>
<p>RColorBrewer</p>
<p>Glimma</p>
<h3 id="VMware-installation"><a href="#VMware-installation" class="headerlink" title="VMware installation:"></a>VMware installation:</h3><p><a href="http://www.zdfans.com/html/5928.html" target="_blank" rel="noopener">vmware-download</a></p>
<p><a href="http://environmentalomics.org/bio-linux-download/" target="_blank" rel="noopener">bio-linux-download</a></p>
<p><a href="http://environmentalomics.org/bio-linux-installation/" target="_blank" rel="noopener">bio-linux-installation</a></p>
<p><a href="https://www.jb51.net/softjc/189144.html" target="_blank" rel="noopener">vmware-tools-installation</a></p>
<p><a href="https://www.cnblogs.com/huangjianxin/p/6343881.html" target="_blank" rel="noopener">filessystem for share</a></p>
<p><a href="https://www.netsarang.com/download/down_form.html?code=622" target="_blank" rel="noopener">X shell-download</a></p>
<p><a href="http://www.bio-info-trainee.com/tag/%E6%9C%8D%E5%8A%A1%E5%99%A8" target="_blank" rel="noopener">other common questions</a></p>
<p><a href="http://blog.sina.com.cn/s/blog_54f07aba0101s1br.html" target="_blank" rel="noopener">R语言编辑器：notepad++</a></p>
<h2 id="Overview"><a href="#Overview" class="headerlink" title="Overview"></a>Overview</h2><ul>
<li>Reading in table of counts</li>
<li>Filtering lowly expressed genes</li>
<li>Quality control</li>
<li>Normalisation for composition bias</li>
<li>Differential expression analysis</li>
<li>Testing relative to a threshold</li>
<li>Visualisation</li>
<li>Gene set testing</li>
</ul>
<h3 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h3><p>There are many steps involved in analysising an RNA-Seq experiment. Analysising an RNASeq experiment begins with sequencing reads. These are aligned to a reference genome, then the number of reads mapped to each gene can be counted. This results in a table of counts, which is what we perform statistical analyses on in R. While mapping and counting are important and necessary tasks, today we will be starting from the count data and getting stuck into analysis.</p>
<h3 id="Data-import"><a href="#Data-import" class="headerlink" title="Data import"></a>Data import</h3><p>Load all the packages will need to analysis the data.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">library(edgeR)</span><br><span class="line">library(limma)</span><br><span class="line">library(Glimma)</span><br><span class="line">library(gplots)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">library(RColorBrewer)</span><br></pre></td></tr></table></figure>
<h3 id="Mouse-mammary-gland-dataset"><a href="#Mouse-mammary-gland-dataset" class="headerlink" title="Mouse mammary gland dataset"></a>Mouse mammary gland dataset</h3><p>The data comes from a Nature Cell Biology paper, <a href="http://www.ncbi.nlm.nih.gov/pubmed/25730472" target="_blank" rel="noopener"><em>EGF-mediated induction of Mcl-1 at the switch to lactation is essential for alveolar cell survival</em></a>. Both the raw data (sequence reads) and processed data (counts) can be downloaded from <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60450" target="_blank" rel="noopener">GSE60450</a>.</p>
<p>This study examines the expression profiles of basal stem-cell enriched cells (B) and committed luminal cells (L) in the mammary gland of virgin, pregnant  and lactating mice. Six groups are present, with one for each combination of cell type and mouse status. Each group contains two biological replicates. We will first use the counts file as a starting point for out analysis. This data has already been aligned to the mouse genome.</p>
<h3 id="Reading-in-the-data"><a href="#Reading-in-the-data" class="headerlink" title="Reading in the data"></a>Reading in the data</h3><p>Download and read in the data.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"># Read the data into R</span><br><span class="line">seqdata &lt;- read.delim(&quot;GSE60450_Lactation-GenewiseCounts.txt&quot;, stringsAsFactors = FALSE)</span><br><span class="line"># Read the sample information into R</span><br><span class="line">sampleinfo &lt;- read.delim(&quot;SampleInfo.txt&quot;)</span><br><span class="line"># Take a look at the data</span><br><span class="line">head(seqdata)</span><br><span class="line"># See how many rows and columns the data frame has</span><br><span class="line">dim(seqdata)</span><br></pre></td></tr></table></figure>
<p>The <code>seqdata</code> object contains information about genes, the first column has the Entrez gene id, the second has the gene length and the remaining columns contain information about the number of reads aligning to the gene in each experimental sample. The sampleinfo file contains basic information about the samples that we will need for the analysis today.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sampleinfo</span><br></pre></td></tr></table></figure>
<p>We will manipulate and reformat the counts matrix into a suitable format for downstream analysis. The first columns in the <code>seqdata</code> dataframe contain annotation information. We need to make a new matrix containing only the counts, but we can store the gene identifiers (the <code>EntrezGeneID</code> column) as rownames. We will add more annotation information about each gene later on in the workshop.</p>
<h3 id="Format-the-data"><a href="#Format-the-data" class="headerlink" title="Format the data"></a>Format the data</h3><p>Create a new data object, <code>countdata</code>, that contains only the counts for the 12 samples.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"># Remove first two columns from seqdata</span><br><span class="line">countdata &lt;- seqdata[,-(1:2)]</span><br><span class="line"># Look at the output</span><br><span class="line">head(countdata)</span><br><span class="line"># Store EntrezGeneID as rownames</span><br><span class="line">rownames(countdata) &lt;- seqdata[,1]</span><br><span class="line"># Take a look at the output</span><br><span class="line">head(countdata)</span><br><span class="line"># Take a look at the column names</span><br><span class="line">colnames(countdata)</span><br></pre></td></tr></table></figure>
<p>The sample names are pretty long, so we’ll shorten these to contain only the relevant information about each sample. We will use the <code>substr</code> command to extract the first 7 characters and use these as the colnames.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"># using substr, you extract the characters starting at position 1 and stopping at position 7 of the colnames</span><br><span class="line">colnames(countdata) &lt;- substr(colnames(countdata),start=1,stop=7)</span><br><span class="line"># Take a look at the output</span><br><span class="line">head(countdata)</span><br></pre></td></tr></table></figure>
<p>The column names are now the same as <code>SampleName</code> in the <code>sampleinfo</code> file, which means the sample information in <code>sampleinfo</code> is in the same order as the columns in  <code>countdata</code>.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">table(colnames(countdata)==sampleinfo$SampleName)</span><br></pre></td></tr></table></figure>
<h3 id="Filtering-to-remove-lowly-expressed-genes"><a href="#Filtering-to-remove-lowly-expressed-genes" class="headerlink" title="Filtering to remove lowly expressed genes"></a>Filtering to remove lowly expressed genes</h3><p><strong>Genes with very low counts</strong> across all libraries provide little evidence for differential expression and they interfere with some of the statistical approximations that are used later. They also add to the multiple testing burden when estimating false discovery rates, reducing power to detect differentially expressed genes. These genes <strong>should be filtered out prior to further analysis</strong>.</p>
<p>There are a few ways to filter out lowly expressed genes. When there are biological replicates in each group, in this case we have a sample size of 2 in each group, we favour filtering on a minimum counts per million threshold present in at least 2 samples. Two represents the smallest sample size for each group in out experiment. In this dataset, we choose to retain genes if they are expressed at a counts-per-million (CPM) above 0.5 in at least two samples.</p>
<p>We will use the <code>cpm</code> function from the <em>edgeR</em> library to generate the CPM values and then filter.</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://cldiao.github.io/RNA-Seq analysis in R/" data-id="cjlkud53a0001vomvegnq6mlh" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/LINUX-VMware-RNA-Seq-R/">LINUX, VMware, RNA-Seq, R</a></li></ul>

    </footer>
  </div>
  
</article>
 


  
    <article id="post-hello-world" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/hello-world/" class="article-date">
  <time datetime="2018-07-24T16:45:45.317Z" itemprop="datePublished">Jul 25, 2018</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/hello-world/">Hello World</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <!-- Table of Contents -->
        
        <p>Welcome to <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a>! This is your very first post. Check <a href="https://hexo.io/docs/" target="_blank" rel="noopener">documentation</a> for more info. If you get any problems when using Hexo, you can find the answer in <a href="https://hexo.io/docs/troubleshooting.html" target="_blank" rel="noopener">troubleshooting</a> or you can ask me on <a href="https://github.com/hexojs/hexo/issues" target="_blank" rel="noopener">GitHub</a>.</p>
<h2 id="Quick-Start"><a href="#Quick-Start" class="headerlink" title="Quick Start"></a>Quick Start</h2><h3 id="Create-a-new-post"><a href="#Create-a-new-post" class="headerlink" title="Create a new post"></a>Create a new post</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo new <span class="string">"My New Post"</span></span><br></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/writing.html" target="_blank" rel="noopener">Writing</a></p>
<h3 id="Run-server"><a href="#Run-server" class="headerlink" title="Run server"></a>Run server</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo server</span><br></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/server.html" target="_blank" rel="noopener">Server</a></p>
<h3 id="Generate-static-files"><a href="#Generate-static-files" class="headerlink" title="Generate static files"></a>Generate static files</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo generate</span><br></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/generating.html" target="_blank" rel="noopener">Generating</a></p>
<h3 id="Deploy-to-remote-sites"><a href="#Deploy-to-remote-sites" class="headerlink" title="Deploy to remote sites"></a>Deploy to remote sites</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo deploy</span><br></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/deployment.html" target="_blank" rel="noopener">Deployment</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://cldiao.github.io/hello-world/" data-id="cjlkud5430004vomvewlo75rf" class="article-share-link">Share</a>
      
      
      
    </footer>
  </div>
  
</article>
 


  
    <article id="post-RNA-Seq入门系列" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/RNA-Seq入门系列/" class="article-date">
  <time datetime="2017-10-17T09:26:41.000Z" itemprop="datePublished">Oct 17, 2017</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/RNA-Seq入门系列/">RNA-Seq入门系列</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <!-- Table of Contents -->
        
        <p>　　前两天电脑崩溃了，无奈恢复了系统，电脑一下子回到了14年的状态，汗！升级win10，安装软件，恢复使用习惯用了两天。得承认没电脑备份习惯让我吃了大亏啊。待我把电脑完全恢复后要再备份一次状态。马上要开会了，学校似乎对网络进行了限制，每天要面对无数的断开连接，确实影响学习的心情，我就趁着这个时间从头复习一下暑假的RNA-Seq入门系列。</p>
<h2 id="RNA-Seq"><a href="#RNA-Seq" class="headerlink" title="RNA-Seq"></a>RNA-Seq</h2><ul>
<li>系统准备</li>
<li>获取测序数据</li>
<li>了解fastq数据</li>
<li>了解参考基因组及基因注释</li>
<li>序列比对</li>
<li>reads计数</li>
<li>差异基因分析</li>
</ul>
<h3 id="系统准备"><a href="#系统准备" class="headerlink" title="系统准备"></a>系统准备</h3><p><a href="https://msdn.microsoft.com/en-us/commandline/wsl/install_guide" target="_blank" rel="noopener">install linux on windows</a>。软件安装git,notepad++,everything，R，Rstudio。</p>
<h3 id="获取测序数据"><a href="#获取测序数据" class="headerlink" title="获取测序数据"></a>获取测序数据</h3><p><a href="https://github.com/CLDIAO/learning-RNA-Seq/blob/master/02-download-data.md" target="_blank" rel="noopener">download data</a></p>
<h3 id="了解fastq数据"><a href="#了解fastq数据" class="headerlink" title="了解fastq数据"></a>了解fastq数据</h3><p><a href="https://github.com/CLDIAO/learning-RNA-Seq/blob/master/03-learn-about-fastq-format%20.md" target="_blank" rel="noopener">learn about fastq fomat</a></p>
<h3 id="了解参考基因组及基因注释"><a href="#了解参考基因组及基因注释" class="headerlink" title="了解参考基因组及基因注释"></a>了解参考基因组及基因注释</h3><p><a href="https://github.com/CLDIAO/learning-RNA-Seq/blob/master/04-learn-about-reference-genome-and-gene-annotation.md" target="_blank" rel="noopener">learn about reference genome and gene annotation</a></p>
<h3 id="序列比对"><a href="#序列比对" class="headerlink" title="序列比对"></a>序列比对</h3><p><a href="https://github.com/CLDIAO/learning-RNA-Seq/blob/master/05-reads-alligenment.md" target="_blank" rel="noopener">reads alignment</a></p>
<h3 id="reads计数"><a href="#reads计数" class="headerlink" title="reads计数"></a>reads计数</h3><p><a href="https://github.com/CLDIAO/learning-RNA-Seq/blob/master/06_count_reads_with_htseq.md" target="_blank" rel="noopener">count reads with HTSeq</a></p>
<h3 id="差异基因分析"><a href="#差异基因分析" class="headerlink" title="差异基因分析"></a>差异基因分析</h3><p><a href="https://github.com/CLDIAO/learning-RNA-Seq/blob/master/07-differential-expression-analysis.md" target="_blank" rel="noopener">differential expression analysis</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://cldiao.github.io/RNA-Seq入门系列/" data-id="cjlkud53v0003vomvj3f8zw4w" class="article-share-link">Share</a>
      
      
      
    </footer>
  </div>
  
</article>
 


  
    <article id="post-markdown编辑器字体字号与颜色" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/markdown编辑器字体字号与颜色/" class="article-date">
  <time datetime="2017-10-11T14:10:28.000Z" itemprop="datePublished">Oct 11, 2017</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/markdown编辑器字体字号与颜色/">markdown编辑器字体字号与颜色</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <p>　　markdown编辑器本身不支持字体、字号、颜色的修改。但因为markdown支持HTML标签，所以可以使用内嵌HTML来实现这些功能。<br>　　使用方法：<br><code>&lt;font size=1 font color=#FF1493 font face=&quot;黑体&quot;&gt;编辑文本&lt;/font&gt;</code>，<br>　　效果：<font size="1" font="" color="#FF1493" face="黑体">编辑文本</font>。<br>　　文本尺寸值：1-7，默认为3。</p>
<p></p><h4>色彩表</h4><br>
        
          <p class="article-more-link">
            <a href="/markdown编辑器字体字号与颜色/#more">继续阅读全文 »</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="https://cldiao.github.io/markdown编辑器字体字号与颜色/" data-id="cjlkud55r000avomvm1yrlsas" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/markdown/">markdown</a></li></ul>

    </footer>
  </div>
  
</article>
 


  
    <article id="post-hexo-github" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/hexo-github/" class="article-date">
  <time datetime="2017-10-08T01:50:37.000Z" itemprop="datePublished">Oct 8, 2017</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/hexo-github/">hexo+github</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <p>　　近期生信技能树公众号推送了一系列我为什么写博客的推文。想到从前并未接触过这个东西，但写博客的好处大家却有共同的见解，就决定去了解一下。在微信群里看到有人说hexo+github的架构，所以花些时间来尝试一下。</p>
<p>　　首先放上我的链接：<a href="https://cldiao.github.io/">cldiao’s blog</a>，10.7下午开始搞，细节的东西在之后慢慢优化吧。<br>
        
          <p class="article-more-link">
            <a href="/hexo-github/#more">继续阅读全文 »</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="https://cldiao.github.io/hexo-github/" data-id="cjlkud55i0008vomvc45qnkhj" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/blog-hexo-github/">blog,hexo,github</a></li></ul>

    </footer>
  </div>
  
</article>
 


  

</section>
           
    <aside id="sidebar">
  
    
  <div class="widget-wrap">
     
        <h3 class="follow-title ">Follow me</h3>
     
    <div class="widget follow">
      
              <a class="github" aria-hidden="true" href="https://github.com/CLDIAO" target="_blank" title="Github"></a>
      
      
      
      
            <a class="email" aria-hidden="true"  href="mailto:dior30102@gmail.com" target="_blank" title="邮箱"></a>
      
    </div>
  </div>


  
    
  
    

  
    
  <div class="widget-wrap">
    <h3 class="widget-title tagcloud">Tag Cloud</h3>
    <div class="widget tagcloud">
      <a href="/tags/LINUX-VMware-RNA-Seq-R/" style="font-size: 14px;">LINUX, VMware, RNA-Seq, R</a> <a href="/tags/Mac-office-Vmware-fusion-Ps/" style="font-size: 14px;">Mac, office, Vmware fusion, Ps</a> <a href="/tags/blog-hexo-github/" style="font-size: 14px;">blog,hexo,github</a> <a href="/tags/markdown/" style="font-size: 14px;">markdown</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title recent-posts">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/Mac上安装一些应用/">Mac上安装一些应用</a>
          </li>
        
          <li>
            <a href="/RNA-Seq analysis in R/">EDITING</a>
          </li>
        
          <li>
            <a href="/hello-world/">Hello World</a>
          </li>
        
          <li>
            <a href="/RNA-Seq入门系列/">RNA-Seq入门系列</a>
          </li>
        
          <li>
            <a href="/markdown编辑器字体字号与颜色/">markdown编辑器字体字号与颜色</a>
          </li>
        
      </ul>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title archive">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/09/">September 2018</a><span class="archive-list-count">1</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/07/">July 2018</a><span class="archive-list-count">2</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2017/10/">October 2017</a><span class="archive-list-count">3</span></li></ul>
    </div>
  </div>


  
    
  
    <!--微信公众号二维码-->

  <div class="widget-wrap">
    <h3 class="follow-title ">WeChat</h3>
    <div class="widget wechat-widget">
        <img src="https://github.com/CLDIAO/CLDIAO.github.io/blob/master/qrcode.jpg" alt="扫码关注" width="250"/>
    </div>
  </div>


  
    
  
</aside>

      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-left">
      &copy; 2014 - 2018 Rumble&nbsp;|&nbsp;
      Theme by <a href="https://github.com/giscafer/hexo-theme-cafe/" target="_blank">Cafe</a>
    </div>
     <div id="footer-right">
      Contact&nbsp;|&nbsp;dior30102@gmail.com
    </div>
  </div>
</footer>
 <script src="/jquery/jquery.min.js"></script>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    <img class="back-to-top-btn" src="/images/fly-to-top.png"/>
<script>
// Elevator script included on the page, already.
window.onload = function() {
  var elevator = new Elevator({
    selector:'.back-to-top-btn',
    element: document.querySelector('.back-to-top-btn'),
    duration: 1000 // milliseconds
  });
}
</script>
      

  
    <script>
      var cloudTieConfig = {
        url: document.location.href, 
        sourceId: "",
        productKey: "e2fb4051c49842688ce669e634bc983f",
        target: "cloud-tie-wrapper"
      };
    </script>
    <script src="https://img1.ws.126.net/f2e/tie/yun/sdk/loader.js"></script>
    

  







<!-- author:forvoid begin -->
<!-- author:forvoid begin -->

<!-- author:forvoid end -->

<!-- author:forvoid end -->


  
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        tex2jax: {
          inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
          processEscapes: true,
          skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
        }
      })
    </script>

    <script type="text/x-mathjax-config">
      MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for (i=0; i < all.length; i += 1) {
          all[i].SourceElement().parentNode.className += ' has-jax';
        }
      })
    </script>
    <script type="text/javascript" src="https://cdn.rawgit.com/mathjax/MathJax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  


 <script src="/js/is.js"></script>


  <link rel="stylesheet" href="/fancybox/jquery.fancybox.css">
  <script src="/fancybox/jquery.fancybox.pack.js"></script>


<script src="/js/script.js"></script>
<script src="/js/elevator.js"></script>
  </div>
</body>
</html>