<!DOCTYPE html>
<html>
<head>
 <meta name="baidu-site-verification" content="d8vuriatOq" />
 <meta name="google-site-verification" content="ZKzrltgUA1iKyVKyH5hNp1TzVg-9LIJqbb09vFyUcWg" />
  <meta charset="utf-8">
  

  
  <title>R语言20题-学习笔记 | Rumble&#39;s blog</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="生信人的20个R语言习题 生信人的20个R语言习题的答案 1.安装一些R包1234source(&quot;https://bioconductor.org/biocLite.R&quot;)biocLite(c(&apos;ALL&apos;,&apos;CLL&apos;,&apos;pasilla&apos;,&apos;clusterProfiler&apos;)) biocLite(c(&apos;airway&apos;,&apos;DESeq2&apos;,&apos;edgeR&apos;,&apos;limma&apos;))install.package">
<meta name="keywords" content="R, 表达矩阵，heatmap, PCA, T检验，火山图，ID转换">
<meta property="og:type" content="article">
<meta property="og:title" content="R语言20题-学习笔记">
<meta property="og:url" content="https://cldiao.github.io/2018/09/27/R语言20题/index.html">
<meta property="og:site_name" content="Rumble&#39;s blog">
<meta property="og:description" content="生信人的20个R语言习题 生信人的20个R语言习题的答案 1.安装一些R包1234source(&quot;https://bioconductor.org/biocLite.R&quot;)biocLite(c(&apos;ALL&apos;,&apos;CLL&apos;,&apos;pasilla&apos;,&apos;clusterProfiler&apos;)) biocLite(c(&apos;airway&apos;,&apos;DESeq2&apos;,&apos;edgeR&apos;,&apos;limma&apos;))install.package">
<meta property="og:locale" content="default">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/boxplot.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/violin.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/histogram.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/density.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/boxplot-2.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/heatmap.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/pheatmap.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/ggplot2.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/heatmap.2.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/upset.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/hclust.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/pca.png">
<meta property="og:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/volcano.png">
<meta property="og:updated_time" content="2018-10-29T01:47:01.151Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="R语言20题-学习笔记">
<meta name="twitter:description" content="生信人的20个R语言习题 生信人的20个R语言习题的答案 1.安装一些R包1234source(&quot;https://bioconductor.org/biocLite.R&quot;)biocLite(c(&apos;ALL&apos;,&apos;CLL&apos;,&apos;pasilla&apos;,&apos;clusterProfiler&apos;)) biocLite(c(&apos;airway&apos;,&apos;DESeq2&apos;,&apos;edgeR&apos;,&apos;limma&apos;))install.package">
<meta name="twitter:image" content="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/boxplot.png">
  
    <link rel="alternate" href="/atom.xml" title="Rumble&#39;s blog" type="application/atom+xml">
  
  
    <link href="/favicon.ico" rel="icon" type="image/x-ico">
    <link rel="icon" href="/favicon.ico">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  <link rel="stylesheet" href="/css/style.css">
</head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">Rumble&#39;s blog</a>
      </h1>
      
        <h2 id="subtitle-wrap">
          <a href="/" id="subtitle">Stay hungry, stay foolish.</a>
        </h2>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
          <a class="main-nav-link" href="/about">About</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="https://cldiao.github.io"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main"><article id="post-R语言20题" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2018/09/27/R语言20题/" class="article-date">
  <time datetime="2018-09-27T13:17:14.000Z" itemprop="datePublished">2018-09-27</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      R语言20题-学习笔记
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
	         <!-- Table of Contents -->
                 
        <p><a href="http://www.bio-info-trainee.com/3409.html" target="_blank" rel="noopener">生信人的20个R语言习题</a></p>
<p><a href="http://www.bio-info-trainee.com/3415.html" target="_blank" rel="noopener">生信人的20个R语言习题的答案</a></p>
<h2 id="1-安装一些R包"><a href="#1-安装一些R包" class="headerlink" title="1.安装一些R包"></a>1.安装一些R包</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">source</span>(<span class="string">"https://bioconductor.org/biocLite.R"</span>)</span><br><span class="line">biocLite(c(<span class="string">'ALL'</span>,<span class="string">'CLL'</span>,<span class="string">'pasilla'</span>,<span class="string">'clusterProfiler'</span>)) </span><br><span class="line">biocLite(c(<span class="string">'airway'</span>,<span class="string">'DESeq2'</span>,<span class="string">'edgeR'</span>,<span class="string">'limma'</span>))</span><br><span class="line">install.packages(<span class="string">"reshape2"</span>, <span class="string">"ggplot2"</span>)</span><br></pre></td></tr></table></figure>
<h2 id="2-了解ExpressionSet对象，比如CLL包里面就有data-sCLLex-，找到它包含的元素，提取其表达矩阵-使用exprs函数-，查看其大小"><a href="#2-了解ExpressionSet对象，比如CLL包里面就有data-sCLLex-，找到它包含的元素，提取其表达矩阵-使用exprs函数-，查看其大小" class="headerlink" title="2.了解ExpressionSet对象，比如CLL包里面就有data(sCLLex) ，找到它包含的元素，提取其表达矩阵(使用exprs函数)，查看其大小"></a>2.了解ExpressionSet对象，比如<code>CLL</code>包里面就有<code>data(sCLLex)</code> ，找到它包含的元素，提取其表达矩阵(使用exprs函数)，查看其大小</h2><a id="more"></a>
<p>   参考：<a href="http://www.bio-info-trainee.com/bioconductor_China/software/limma.html" target="_blank" rel="noopener">http://www.bio-info-trainee.com/bioconductor_China/software/limma.html</a></p>
<p>   参考：<a href="https://github.com/bioconductor-china/basic/blob/master/ExpressionSet.md" target="_blank" rel="noopener">https://github.com/bioconductor-china/basic/blob/master/ExpressionSet.md</a></p>
   <figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(CLL) </span><br><span class="line"><span class="comment"># load package</span></span><br><span class="line">?CLL </span><br><span class="line"><span class="comment"># help</span></span><br><span class="line">data(CLL) </span><br><span class="line"><span class="comment"># load dataset</span></span><br><span class="line">sCLLex </span><br><span class="line"><span class="comment"># check the dataset: ExprssionSet</span></span><br><span class="line"><span class="comment"># ExpressionSet: Container for high-throughput assays and experimental metadata. The class is derived from eSet,and requires a matrix named exprs as assayData member.</span></span><br><span class="line">exprSet &lt;- exprs(sCLLex) </span><br><span class="line">dim(exprSet)</span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">5</span>,<span class="number">1</span>:<span class="number">5</span>]</span><br><span class="line"><span class="comment"># exprs: These generic functions access the expression and error measurements of assay data stored in an object derived from the eSet-class.</span></span><br><span class="line"><span class="comment"># [1] 12625    22</span></span><br><span class="line">samples &lt;- sampleNames(sCLLex)</span><br><span class="line"><span class="comment"># sampleNames: These generic functions access the feature names (typically, gene or SNP identifiers) and sample names stored in an object derived from the eSet-class.</span></span><br><span class="line">pdata &lt;- pData(sCLLex)</span><br><span class="line"><span class="comment"># pData: These generic functions access the phenotypic data (e.g., covariates) and meta-data (e.g., descriptions of covariates) associated with an experiment.</span></span><br><span class="line">pdata</span><br><span class="line"><span class="comment"># check the pdata</span></span><br><span class="line">group_list &lt;- as.character(pdata[,<span class="number">2</span>])</span><br></pre></td></tr></table></figure>
<h2 id="3-了解-str-head-help函数，作用于-第二步提取到的表达矩阵"><a href="#3-了解-str-head-help函数，作用于-第二步提取到的表达矩阵" class="headerlink" title="3.了解 str,head,help函数，作用于 第二步提取到的表达矩阵"></a>3.了解 str,head,help函数，作用于 第二步提取到的表达矩阵</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">str(exprSet)</span><br><span class="line"><span class="comment"># str: Compactly display the internal structure of an R object, a diagnostic function and an alternative to summary (and to some extent, dput).</span></span><br><span class="line">head(exprSet)</span><br></pre></td></tr></table></figure>
<h2 id="4-安装并了解-hgu95av2-db-包-看看-ls-quot-package-hgu95av2-db-quot-后-显示的那些变量"><a href="#4-安装并了解-hgu95av2-db-包-看看-ls-quot-package-hgu95av2-db-quot-后-显示的那些变量" class="headerlink" title="4.安装并了解 hgu95av2.db 包,看看 ls(&quot;package:hgu95av2.db&quot;) 后 显示的那些变量"></a>4.安装并了解 <code>hgu95av2.db</code> 包,看看 <code>ls(&quot;package:hgu95av2.db&quot;)</code> 后 显示的那些变量</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">source</span>(<span class="string">"https://bioconductor.org/biocLite.R"</span>)</span><br><span class="line">biocLite(<span class="string">"hgu95av2.db"</span>)</span><br><span class="line">??hgu95av2.db</span><br><span class="line"><span class="keyword">library</span>(hgu95av2.db)</span><br><span class="line">ls(<span class="string">"package:hgu95av2.db"</span>)</span><br><span class="line">columns(hgu95av2.db)</span><br><span class="line"><span class="comment"># columns shows which kinds of data can be returned for the AnnotationDb object.</span></span><br></pre></td></tr></table></figure>
<h2 id="5-理解-head-toTable-hgu95av2SYMBOL-的用法，找到-TP53-基因对应的探针ID"><a href="#5-理解-head-toTable-hgu95av2SYMBOL-的用法，找到-TP53-基因对应的探针ID" class="headerlink" title="5.理解 head(toTable(hgu95av2SYMBOL)) 的用法，找到 TP53 基因对应的探针ID"></a>5.理解 <code>head(toTable(hgu95av2SYMBOL))</code> 的用法，找到 TP53 基因对应的探针ID</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">head(toTable(hgu95av2SYMBOL))</span><br><span class="line"><span class="comment"># toTable: What we usually call "annotation maps" are in fact Bimap objects. Extract all the columns of the map</span></span><br><span class="line"><span class="comment">#   probe_id  symbol</span></span><br><span class="line"><span class="number">1</span>   1000_at   MAPK3</span><br><span class="line"><span class="number">2</span>   1001_at    TIE1</span><br><span class="line"><span class="number">3</span> 1002_f_at CYP2C19</span><br><span class="line"><span class="number">4</span> 1003_s_at   CXCR5</span><br><span class="line"><span class="number">5</span>   1004_at   CXCR5</span><br><span class="line"><span class="number">6</span>   1005_at   DUSP1</span><br><span class="line">ids &lt;- toTable(hgu95av2SYMBOL)</span><br><span class="line">View(ids)</span><br><span class="line">TP53 &lt;- c(<span class="string">"1939_at"</span>, <span class="string">"1974_s_at"</span>, <span class="string">"31618_at"</span>)</span><br></pre></td></tr></table></figure>
<h2 id="6-理解探针与基因的对应关系，总共多少个基因，基因最多对应多少个探针，是哪些基因，是不是因为这些基因很长，所以在其上面设计多个探针呢？"><a href="#6-理解探针与基因的对应关系，总共多少个基因，基因最多对应多少个探针，是哪些基因，是不是因为这些基因很长，所以在其上面设计多个探针呢？" class="headerlink" title="6.理解探针与基因的对应关系，总共多少个基因，基因最多对应多少个探针，是哪些基因，是不是因为这些基因很长，所以在其上面设计多个探针呢？"></a>6.理解探针与基因的对应关系，总共多少个基因，基因最多对应多少个探针，是哪些基因，是不是因为这些基因很长，所以在其上面设计多个探针呢？</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">length(unique(ids$symbol))</span><br><span class="line"><span class="comment"># [1] 8585</span></span><br><span class="line"><span class="comment"># check the number of occurrences of every symbol in ids</span></span><br><span class="line"><span class="comment"># unique: returns a vector, data frame or array like x but with duplicate elements/rows removed.</span></span><br><span class="line">tail(sort(table(ids$symbol)))</span><br><span class="line"><span class="comment"># sort: Sort (or order) a vector or factor (partially) into ascending or descending order. </span></span><br><span class="line">plot(tail(sort(table(ids$symbol))))</span><br><span class="line"><span class="comment"># YME1L1  GAPDH INPP4A    MYB PTGER3  STAT1 </span></span><br><span class="line">   <span class="number">7</span>      <span class="number">8</span>      <span class="number">8</span>      <span class="number">8</span>      <span class="number">8</span>      <span class="number">8</span></span><br></pre></td></tr></table></figure>
<p><a href="http://www.affymetrix.com/support/help/faqs/mouse_430/faq_8.affx" target="_blank" rel="noopener"><em>What is a probe set? What do the different suffixes attached to a probe set name mean?</em></a></p>
<p><a href="https://www.ohsu.edu/xd/research/research-cores/gene-profiling-shared-resource/project-design/array-technology/affymetrix-genechip-arrays.cfm" target="_blank" rel="noopener">Probe Set Extension Nomenclature</a></p>
<h2 id="7（8）-第二步提取到的表达矩阵是12625个探针在22个样本的表达量矩阵，找到那些不在-hgu95av2-db-包收录的对应着SYMBOL的探针。"><a href="#7（8）-第二步提取到的表达矩阵是12625个探针在22个样本的表达量矩阵，找到那些不在-hgu95av2-db-包收录的对应着SYMBOL的探针。" class="headerlink" title="7（8）.第二步提取到的表达矩阵是12625个探针在22个样本的表达量矩阵，找到那些不在 hgu95av2.db 包收录的对应着SYMBOL的探针。"></a>7（8）.第二步提取到的表达矩阵是12625个探针在22个样本的表达量矩阵，找到那些不在 <code>hgu95av2.db</code> 包收录的对应着SYMBOL的探针。</h2><p>过滤表达矩阵，删除那1165个没有对应基因名字的探针。 </p>
<p>提示：有1165个探针是没有对应基因名字的。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">table(rownames(exprSet)) %<span class="keyword">in</span>% ids$probe_id</span><br><span class="line"><span class="comment"># %in% is a more intuitive interface as a binary operator, which returns a logical vector indicating if there is a match or not for its left operand.</span></span><br><span class="line">n_exprSet &lt;- exprSet[!(rownames(exprSet) %<span class="keyword">in</span>% ids$probe_id),]</span><br><span class="line">dim(n_exprSet)</span><br><span class="line"><span class="comment"># [1] 1165   22</span></span><br><span class="line">View(n_exprSet)</span><br><span class="line"><span class="comment"># These probes are not in the package.</span></span><br><span class="line">exprSet &lt;- exprSet[rownames(exprSet) %<span class="keyword">in</span>% ids$probe_id),]</span><br><span class="line">dim(exprSet)</span><br><span class="line">[<span class="number">1</span>] <span class="number">11460</span>    <span class="number">22</span></span><br><span class="line">View(exprSet)</span><br><span class="line"><span class="comment"># These probes are in the package.</span></span><br></pre></td></tr></table></figure>
<h2 id="9-整合表达矩阵，多个探针对应一个基因的情况下，只保留在所有样本里面平均表达量最大的那个探针。"><a href="#9-整合表达矩阵，多个探针对应一个基因的情况下，只保留在所有样本里面平均表达量最大的那个探针。" class="headerlink" title="9.整合表达矩阵，多个探针对应一个基因的情况下，只保留在所有样本里面平均表达量最大的那个探针。"></a>9.整合表达矩阵，多个探针对应一个基因的情况下，只保留在所有样本里面平均表达量最大的那个探针。</h2><p>提示，理解 tapply,by,aggregate,split 函数 , 首先对每个基因找到最大表达量的探针。</p>
<p>然后根据得到探针去过滤原始表达矩阵</p>
<h2 id="10-把过滤后的表达矩阵更改行名为基因的symbol，因为这个时候探针和基因是一对一关系了。"><a href="#10-把过滤后的表达矩阵更改行名为基因的symbol，因为这个时候探针和基因是一对一关系了。" class="headerlink" title="10.把过滤后的表达矩阵更改行名为基因的symbol，因为这个时候探针和基因是一对一关系了。"></a>10.把过滤后的表达矩阵更改行名为基因的symbol，因为这个时候探针和基因是一对一关系了。</h2>   <figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># tapply: Apply a function to each cell of a ragged array, that is to each (non-empty) group of values given by a unique combination of the levels of certain factors.</span></span><br><span class="line"><span class="comment"># by: is an object-oriented wrapper for tapply applied to data frames.</span></span><br><span class="line"><span class="comment"># aggregate: Given an environment and an aggregator (an object of class aggregate simple aggregations are made.</span></span><br><span class="line"><span class="comment"># split: divides the data in the vector x into the groups defined by f. The replacement forms replace values corresponding to such a division. unsplit reverses the effect of split.</span></span><br><span class="line"><span class="comment"># match: Returns a vector of the positions of (first) matches of its first argument in its second.</span></span><br><span class="line"><span class="keyword">library</span>(reshape2)</span><br><span class="line">rownames(exprSet) &lt;- ids[match(rownames(exprSet), ids$probe_id), <span class="number">2</span>]</span><br><span class="line">exprSet_L &lt;- melt(exprSet)</span><br><span class="line"><span class="comment"># melt() Convert an object into a molten data frame. </span></span><br><span class="line">View(exprSet_L)</span><br><span class="line"><span class="comment">#      Var1      Var2    value</span></span><br><span class="line"><span class="number">1</span>   MAPK3 CLL11.CEL <span class="number">5.743132</span></span><br><span class="line"><span class="number">2</span>    TIE1 CLL11.CEL <span class="number">2.285143</span></span><br><span class="line"><span class="number">3</span> CYP2C19 CLL11.CEL <span class="number">3.309294</span></span><br><span class="line"><span class="number">4</span>   CXCR5 CLL11.CEL <span class="number">1.085264</span></span><br><span class="line"><span class="number">5</span>   CXCR5 CLL11.CEL <span class="number">7.544884</span></span><br><span class="line"><span class="number">6</span>   DUSP1 CLL11.CEL <span class="number">5.083793</span></span><br><span class="line">colnames(exprSet_L) &lt;- c(<span class="string">'probe'</span>, <span class="string">'sample'</span>, <span class="string">'value'</span>)</span><br><span class="line">View(exprSet_L)</span><br><span class="line"><span class="comment">#      Var1      Var2    value</span></span><br><span class="line"><span class="number">1</span>   MAPK3 CLL11.CEL <span class="number">5.743132</span></span><br><span class="line"><span class="number">2</span>    TIE1 CLL11.CEL <span class="number">2.285143</span></span><br><span class="line"><span class="number">3</span> CYP2C19 CLL11.CEL <span class="number">3.309294</span></span><br><span class="line"><span class="number">4</span>   CXCR5 CLL11.CEL <span class="number">1.085264</span></span><br><span class="line"><span class="number">5</span>   CXCR5 CLL11.CEL <span class="number">7.544884</span></span><br><span class="line"><span class="number">6</span>   DUSP1 CLL11.CEL <span class="number">5.083793</span></span><br><span class="line">exprSet_L$group &lt;- rep(group_list,each=nrow(exprSet))</span><br><span class="line"><span class="comment"># add a column named group</span></span><br><span class="line">View(exprSet_L)</span><br><span class="line">     Var1      Var2    value    group</span><br><span class="line"><span class="number">1</span>   MAPK3 CLL11.CEL <span class="number">5.743132</span> progres.</span><br><span class="line"><span class="number">2</span>    TIE1 CLL11.CEL <span class="number">2.285143</span> progres.</span><br><span class="line"><span class="number">3</span> CYP2C19 CLL11.CEL <span class="number">3.309294</span> progres.</span><br><span class="line"><span class="number">4</span>   CXCR5 CLL11.CEL <span class="number">1.085264</span> progres.</span><br><span class="line"><span class="number">5</span>   CXCR5 CLL11.CEL <span class="number">7.544884</span> progres.</span><br><span class="line"><span class="number">6</span>   DUSP1 CLL11.CEL <span class="number">5.083793</span> progres.</span><br></pre></td></tr></table></figure>
<p><a href="https://www.r-bloggers.com/melt/" target="_blank" rel="noopener">melt</a></p>
<h2 id="11-对第10步得到的表达矩阵进行探索，先画第一个样本的所有基因的表达量的boxplot-hist-density-，-然后画所有样本的-这些图"><a href="#11-对第10步得到的表达矩阵进行探索，先画第一个样本的所有基因的表达量的boxplot-hist-density-，-然后画所有样本的-这些图" class="headerlink" title="11.对第10步得到的表达矩阵进行探索，先画第一个样本的所有基因的表达量的boxplot,hist,density ， 然后画所有样本的 这些图"></a>11.对第10步得到的表达矩阵进行探索，先画第一个样本的所有基因的表达量的boxplot,hist,density ， 然后画所有样本的 这些图</h2><p>理解ggplot2的绘图语法，数据和图形元素的映射关系</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line">p &lt;- ggplot(exprSet_L,aes(x=sample, y=value, fill=group)) +</span><br><span class="line">  geom_boxplot()</span><br><span class="line">  </span><br><span class="line">p</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/boxplot.png" alt=""></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># boxplot</span></span><br><span class="line">p &lt;- ggplot(exprSet_L,aes(x=sample, y=value, fill=group))+ </span><br><span class="line">  geom_violin()</span><br><span class="line">p</span><br><span class="line"><span class="comment"># violin</span></span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/violin.png" alt=""></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">p &lt;- ggplot(exprSet_L, aes(value, fill=group))+</span><br><span class="line">  geom_histogram(bins = <span class="number">200</span>)+</span><br><span class="line">  facet_wrap(~sample, nrow = <span class="number">4</span>)</span><br><span class="line">p</span><br><span class="line"><span class="comment"># density</span></span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/histogram.png" alt=""></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">p &lt;- ggplot(exprSet_L,aes(value,col=group))+</span><br><span class="line">  geom_density()</span><br><span class="line">p</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/density.png" alt=""></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">p=ggplot(exprSet_L,aes(x=sample,y=value,fill=group))+geom_boxplot()</span><br><span class="line">p=p+stat_summary(fun.y=<span class="string">"mean"</span>,geom=<span class="string">"point"</span>,shape=<span class="number">23</span>,size=<span class="number">3</span>,fill=<span class="string">"red"</span>)</span><br><span class="line">p=p+theme_set(theme_set(theme_bw(base_size=<span class="number">20</span>)))</span><br><span class="line">p=p+theme(text=element_text(face=<span class="string">'bold'</span>),axis.text.x=element_text(angle=<span class="number">30</span>,hjust=<span class="number">1</span>),axis.title=element_blank())</span><br><span class="line">print(p)</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/boxplot-2.png" alt=""></p>
<p>参考：<a href="http://bio-info-trainee.com/tmp/basic_visualization_for_expression_matrix.html" target="_blank" rel="noopener">basic visualization for expression matrix</a></p>
<h2 id="12-理解统计学指标mean-median-max-min-sd-var-mad并计算出每个基因在所有样本的这些统计学指标，最后按照mad值排序，取top-50-mad值的基因，得到列表。"><a href="#12-理解统计学指标mean-median-max-min-sd-var-mad并计算出每个基因在所有样本的这些统计学指标，最后按照mad值排序，取top-50-mad值的基因，得到列表。" class="headerlink" title="12.理解统计学指标mean,median,max,min,sd,var,mad并计算出每个基因在所有样本的这些统计学指标，最后按照mad值排序，取top 50 mad值的基因，得到列表。"></a>12.理解统计学指标mean,median,max,min,sd,var,mad并计算出每个基因在所有样本的这些统计学指标，最后按照mad值排序，取top 50 mad值的基因，得到列表。</h2><p>注意：这个题目出的并不合规，请仔细看。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">apply() Returns a vector or array or list of values obtained by applying a <span class="keyword">function</span> to margins of an array or matrix.  <span class="number">1</span> indicates rows, <span class="number">2</span> indicates columns, c(<span class="number">1</span>, <span class="number">2</span>) indicates rows and columns.</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># mean</span></span><br><span class="line">g_mean &lt;- tail(sort(apply(exprSet, <span class="number">1</span>, mean)),<span class="number">50</span>)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># median</span></span><br><span class="line">g_median &lt;-  tail(sort(apply(exprSet, <span class="number">1</span>, median)),<span class="number">50</span>)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># max</span></span><br><span class="line">g_max &lt;- tail(sort(apply(exprSet, <span class="number">1</span>, max)),<span class="number">50</span>)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># min </span></span><br><span class="line">g_min &lt;- tail(sort(apply(exprSet, <span class="number">1</span>, min)),<span class="number">50</span>)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># sd</span></span><br><span class="line">g_sd &lt;- tail(sort(apply(exprSet, <span class="number">1</span>, sd)),<span class="number">50</span>)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># var</span></span><br><span class="line">g_var &lt;- tail(sort(apply(exprSet, <span class="number">1</span>, var)),<span class="number">50</span>)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># mad: Compute the median absolute deviation.</span></span><br><span class="line">g_mad &lt;- tail(sort(apply(exprSet, <span class="number">1</span>, mad)),<span class="number">50</span>)</span><br></pre></td></tr></table></figure>
<h2 id="13-根据第12步骤得到top-50-mad值的基因列表来取表达矩阵的子集，并且热图可视化子表达矩阵。试试看其它5种热图的包的不同效果。"><a href="#13-根据第12步骤得到top-50-mad值的基因列表来取表达矩阵的子集，并且热图可视化子表达矩阵。试试看其它5种热图的包的不同效果。" class="headerlink" title="13.根据第12步骤得到top 50 mad值的基因列表来取表达矩阵的子集，并且热图可视化子表达矩阵。试试看其它5种热图的包的不同效果。"></a>13.根据第12步骤得到top 50 mad值的基因列表来取表达矩阵的子集，并且热图可视化子表达矩阵。试试看其它5种热图的包的不同效果。</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">g_mad</span><br><span class="line"><span class="comment">#    DUSP5   IGFBP4     H1FX    ENPP2     FLNA   CLEC2B   TSPYL2   ZNF266   S100A9    NR4A2    TGFBI     ARF6    APBB2 </span></span><br><span class="line"><span class="number">1.553782</span> <span class="number">1.554888</span> <span class="number">1.557412</span> <span class="number">1.561485</span> <span class="number">1.574436</span> <span class="number">1.578055</span> <span class="number">1.582430</span> <span class="number">1.587748</span> <span class="number">1.608285</span> <span class="number">1.612875</span> <span class="number">1.643149</span> <span class="number">1.654548</span> <span class="number">1.674443</span> </span><br><span class="line">    VCAN    RBM38     CAPG   PLXNC1     RGS2   RNASE6    VAMP5     CYBB     GNLY     CCL3     OAS1    ENPP2    TRIB2 </span><br><span class="line"><span class="number">1.681098</span> <span class="number">1.703638</span> <span class="number">1.713747</span> <span class="number">1.718297</span> <span class="number">1.770151</span> <span class="number">1.775430</span> <span class="number">1.791017</span> <span class="number">1.811973</span> <span class="number">1.814364</span> <span class="number">1.862311</span> <span class="number">1.883509</span> <span class="number">1.909442</span> <span class="number">1.937294</span> </span><br><span class="line"> ZNF804A     H1FX      IGH     JUND  SLC25A1    PCDH9    VIPR1   COBLL1  GUSBP11   S100A8      HBB      FOS   LHFPL2 </span><br><span class="line"><span class="number">1.986163</span> <span class="number">2.052788</span> <span class="number">2.090866</span> <span class="number">2.105234</span> <span class="number">2.115238</span> <span class="number">2.144223</span> <span class="number">2.171912</span> <span class="number">2.179666</span> <span class="number">2.204212</span> <span class="number">2.220970</span> <span class="number">2.267136</span> <span class="number">2.268946</span> <span class="number">2.317045</span> </span><br><span class="line">    FCN1    ZAP70    IGLC1   LGALS1      HBB      FOS   SLAMF1     TCF7      DMD  IGF2BP3   FAM30A </span><br><span class="line"><span class="number">2.371590</span> <span class="number">2.579046</span> <span class="number">2.590895</span> <span class="number">2.600604</span> <span class="number">2.732776</span> <span class="number">2.938078</span> <span class="number">2.944105</span> <span class="number">2.993352</span> <span class="number">3.071541</span> <span class="number">3.234011</span> <span class="number">3.982191</span> </span><br><span class="line">choose_gene &lt;- names(g_mad)</span><br><span class="line">choose_matrix &lt;- exprSet[choose_gene,]</span><br></pre></td></tr></table></figure>
<p>热图的几种画法：</p>
<h3 id="heatmap"><a href="#heatmap" class="headerlink" title="heatmap"></a>heatmap</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">choose_matrix &lt;- t(scale(t(choose_matrix)))</span><br><span class="line"><span class="comment"># t() Given a rectangular object x, t returns the transpose of x.</span></span><br><span class="line"><span class="comment"># scale is generic function whose default method centers and/or scales the columns of a numeric matrix.</span></span><br><span class="line"><span class="comment"># The value of scale determines how column scaling is performed.</span></span><br><span class="line">rc &lt;- rainbow(nrow(choose_matrix), start = <span class="number">0</span>, end = <span class="number">.3</span>)</span><br><span class="line">cc &lt;- rainbow(ncol(choose_matrix), start = <span class="number">0</span>, end = <span class="number">.3</span>)</span><br><span class="line"><span class="comment"># the heatmap doesnot offer the color key</span></span><br><span class="line">heatmap(choose_matrix, </span><br><span class="line">        col = cm.colors(<span class="number">256</span>), <span class="comment"># https://www.r-bloggers.com/color-palettes-in-r/锛?</span></span><br><span class="line">        RowSideColors = rc, ColSideColors = cc,<span class="comment"># color names for a horizontal side bar </span></span><br><span class="line">        <span class="comment">#Colv = NA, Rowv = NA # determines if and how the row dendrogram should be computed and reordered</span></span><br><span class="line">        margins = c(<span class="number">6</span>, <span class="number">5</span>), <span class="comment"># margins for column and row names</span></span><br><span class="line">        xlab = <span class="string">"samples"</span>, ylab = <span class="string">"genes"</span>, <span class="comment"># x/y axis titles</span></span><br><span class="line">        main = <span class="string">"heatmap"</span>, <span class="comment"># titles</span></span><br><span class="line">        scale = <span class="string">"column"</span></span><br><span class="line">        )</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/heatmap.png" alt=""></p>
<h3 id="pheatmap"><a href="#pheatmap" class="headerlink" title="pheatmap"></a>pheatmap</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(pheatmap)</span><br><span class="line">pheatmap(choose_matrix</span><br><span class="line">         <span class="comment"># kmeans_k = NA, # the number of kmeans clusters to make</span></span><br><span class="line">         <span class="comment"># clustering_distance_rows = "correlation", # distance measure used in clustering rows. </span></span><br><span class="line">         <span class="comment"># cluster_rows = F, cluster_cols = F, #  if rows/cols should be clustered</span></span><br><span class="line">         <span class="comment"># legend = F, # logical to determine if legend should be drawn or not.</span></span><br><span class="line">         <span class="comment"># cellwidth = 12, cellheight = 15,</span></span><br><span class="line">         <span class="comment"># filename = "test.pdf"锛? # file path where to save the picture</span></span><br><span class="line">         )</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/pheatmap.png" alt=""></p>
<h3 id="ggplot2"><a href="#ggplot2" class="headerlink" title="ggplot2"></a>ggplot2</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(ggplot2)</span><br><span class="line">choose_matrix_m &lt;- melt(choose_matrix)</span><br><span class="line">p &lt;- ggplot(choose_matrix_m, aes(x= Var2, y = Var1))</span><br><span class="line">p &lt;- p + geom_tile(aes(fill=value))</span><br><span class="line">p</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/ggplot2.png" alt=""></p>
<p>参考：<a href="https://www.cnblogs.com/xiaofeiIDO/p/8337912.html" target="_blank" rel="noopener">ggplot 绘制热图</a></p>
<h3 id="heatmap-2"><a href="#heatmap-2" class="headerlink" title="heatmap.2"></a>heatmap.2</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># heatmap.2 offers the color key.</span></span><br><span class="line"><span class="keyword">library</span>(gplots)</span><br><span class="line">heatmap.2 (choose_matrix)</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/heatmap.2.png" alt=""></p>
<h3 id="lattice"><a href="#lattice" class="headerlink" title="lattice"></a>lattice</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(latticeExtra)</span><br><span class="line">?levelplot</span><br></pre></td></tr></table></figure>
<h2 id="14-取不同统计学指标mean-median-max-mean-sd-var-mad的各top50基因列表，使用UpSetR包来看他们之间的overlap情况。"><a href="#14-取不同统计学指标mean-median-max-mean-sd-var-mad的各top50基因列表，使用UpSetR包来看他们之间的overlap情况。" class="headerlink" title="14.取不同统计学指标mean,median,max,mean,sd,var,mad的各top50基因列表，使用UpSetR包来看他们之间的overlap情况。"></a>14.取不同统计学指标mean,median,max,mean,sd,var,mad的各top50基因列表，使用UpSetR包来看他们之间的overlap情况。</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(UpSetR)</span><br><span class="line">??UpSetR</span><br><span class="line"><span class="comment"># Visualization of set intersections using novel UpSet matrix design.</span></span><br></pre></td></tr></table></figure>
<p><a href="https://ytlogos.github.io/2017/10/15/R%E8%AF%AD%E8%A8%80%E5%8F%AF%E8%A7%86%E5%8C%96%E4%B9%8BUpSetR%E5%8C%85/" target="_blank" rel="noopener">R语言可视化之UpSetR包</a></p>
<p><a href="https://cran.r-project.org/web/packages/UpSetR/README.html" target="_blank" rel="noopener">UpSetR - CRAN.R-project.org</a></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">g_all &lt;- unique(c(names(g_mean), </span><br><span class="line">                  names(g_median), </span><br><span class="line">                  names(g_max), </span><br><span class="line">                  names(g_min), </span><br><span class="line">                  names(g_sd), </span><br><span class="line">                  names(g_var), </span><br><span class="line">                  names(g_mad)</span><br><span class="line">                  ))</span><br><span class="line">dat &lt;- data.frame(g_all,</span><br><span class="line">                  g_mean = ifelse(g_all %<span class="keyword">in</span>% names(g_mean), <span class="number">1</span>, <span class="number">0</span>),</span><br><span class="line">                  g_median = ifelse(g_all%<span class="keyword">in</span>% names(g_median), <span class="number">1</span> ,<span class="number">0</span>),</span><br><span class="line">                  g_max = ifelse(g_all %<span class="keyword">in</span>% names(g_max), <span class="number">1</span>, <span class="number">0</span>),</span><br><span class="line">                  g_min = ifelse(g_all %<span class="keyword">in</span>% names(g_min), <span class="number">1</span>, <span class="number">0</span>),</span><br><span class="line">                  g_sd = ifelse(g_all %<span class="keyword">in</span>% names(g_sd), <span class="number">1</span>, <span class="number">0</span>),</span><br><span class="line">                  g_var = ifelse(g_all %<span class="keyword">in</span>% names(g_var), <span class="number">1</span>, <span class="number">0</span>),</span><br><span class="line">                  g_mad = ifelse(g_all %<span class="keyword">in</span>% names(g_mad), <span class="number">1</span>, <span class="number">0</span>)</span><br><span class="line">                  )</span><br><span class="line">upset(dat, nsets = <span class="number">7</span>)</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/upset.png" alt=""></p>
<h2 id="15-在第二步的基础上面提取CLL包里面的data-sCLLex-数据对象的样本的表型数据。"><a href="#15-在第二步的基础上面提取CLL包里面的data-sCLLex-数据对象的样本的表型数据。" class="headerlink" title="15.在第二步的基础上面提取CLL包里面的data(sCLLex) 数据对象的样本的表型数据。"></a>15.在第二步的基础上面提取<code>CLL</code>包里面的<code>data(sCLLex)</code> 数据对象的样本的表型数据。</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(CLL)</span><br><span class="line">data(sCLLex)</span><br><span class="line">pdata &lt;- pData(sCLLex)</span><br><span class="line">group_list &lt;- as.character(pdata[,<span class="number">2</span>])</span><br></pre></td></tr></table></figure>
<h2 id="16-对所有样本的表达矩阵进行聚类并且绘图，然后添加样本的临床表型数据信息-更改样本名"><a href="#16-对所有样本的表达矩阵进行聚类并且绘图，然后添加样本的临床表型数据信息-更改样本名" class="headerlink" title="16. 对所有样本的表达矩阵进行聚类并且绘图，然后添加样本的临床表型数据信息(更改样本名)"></a>16. 对所有样本的表达矩阵进行聚类并且绘图，然后添加样本的临床表型数据信息(更改样本名)</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">dim(exprSet)</span><br><span class="line"><span class="comment"># [1] 11460    22</span></span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">5</span>,<span class="number">1</span>:<span class="number">5</span>]</span><br><span class="line"><span class="comment">#        CLL11.CEL CLL12.CEL CLL13.CEL CLL14.CEL CLL15.CEL</span></span><br><span class="line">MAPK3    <span class="number">5.743132</span>  <span class="number">6.219412</span>  <span class="number">5.523328</span>  <span class="number">5.340477</span>  <span class="number">5.229904</span></span><br><span class="line">TIE1     <span class="number">2.285143</span>  <span class="number">2.291229</span>  <span class="number">2.287986</span>  <span class="number">2.295313</span>  <span class="number">2.662170</span></span><br><span class="line">CYP2C19  <span class="number">3.309294</span>  <span class="number">3.318466</span>  <span class="number">3.354423</span>  <span class="number">3.327130</span>  <span class="number">3.365113</span></span><br><span class="line">CXCR5    <span class="number">1.085264</span>  <span class="number">1.117288</span>  <span class="number">1.084010</span>  <span class="number">1.103217</span>  <span class="number">1.074243</span></span><br><span class="line">CXCR5    <span class="number">7.544884</span>  <span class="number">7.671801</span>  <span class="number">7.474025</span>  <span class="number">7.152482</span>  <span class="number">6.902932</span></span><br><span class="line">colnames(exprSet) &lt;- paste(group_list,<span class="number">1</span>:<span class="number">22</span>,sep=<span class="string">''</span>)</span><br><span class="line"><span class="comment"># change the sample names</span></span><br><span class="line">nodePar &lt;- list(lab.cex = <span class="number">0.6</span>, pch = c(<span class="number">10</span>, <span class="number">15</span>), cex = <span class="number">0.6</span>, col = <span class="string">"green"</span>)</span><br><span class="line"><span class="comment"># Define nodePar,lab.cex </span></span><br><span class="line">hc=hclust(dist(t(exprSet)))</span><br><span class="line"><span class="comment"># hclust() Hierarchical cluster analysis on a set of dissimilarities and methods for analyzing it.</span></span><br><span class="line"><span class="comment"># by default the row names or row numbers of the original data are used for the leaves of the tree. </span></span><br><span class="line">par(mar=c(<span class="number">5</span>,<span class="number">5</span>,<span class="number">5</span>,<span class="number">10</span>))</span><br><span class="line"><span class="comment"># mar() A numerical vector of the form c(bottom, left, top, right) which gives the number of lines of margin to be specified on the four sides of the plot. The default is c(5, 4, 4, 2) + 0.1.</span></span><br><span class="line"><span class="comment"># par() can be used to set or query graphical parameters.</span></span><br><span class="line">plot(as.dendrogram(hc), nodePar = nodePar, horiz = <span class="literal">TRUE</span>)</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/hclust.png" alt=""></p>
<h2 id="17-对所有样本的表达矩阵进行PCA分析并且绘图，同样要添加表型信息。"><a href="#17-对所有样本的表达矩阵进行PCA分析并且绘图，同样要添加表型信息。" class="headerlink" title="17. 对所有样本的表达矩阵进行PCA分析并且绘图，同样要添加表型信息。"></a>17. 对所有样本的表达矩阵进行PCA分析并且绘图，同样要添加表型信息。</h2><p><a href="https://zhuanlan.zhihu.com/p/28172816" target="_blank" rel="noopener">Learn R | 可视化之ggfortify包</a></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">install.packages(<span class="string">"ggfortify"</span>)</span><br><span class="line"><span class="keyword">library</span>(ggfortify)</span><br><span class="line">exprSet &lt;- exprs(sCLLex)</span><br><span class="line">df &lt;- as.data.frame(t(exprSet))</span><br><span class="line">df$group &lt;- group_list</span><br><span class="line"><span class="comment"># autoplot uses ggplot2 to draw a particular plot for an object of a particular class in a single command.</span></span><br><span class="line">autoplot(prcomp(df[,<span class="number">1</span>:(ncol(df)-<span class="number">1</span>)]), data=df, colour = <span class="string">'group'</span>)</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/pca.png" alt=""></p>
<h2 id="18-根据表达矩阵及样本分组信息进行批量T检验，得到检验结果表格"><a href="#18-根据表达矩阵及样本分组信息进行批量T检验，得到检验结果表格" class="headerlink" title="18. 根据表达矩阵及样本分组信息进行批量T检验，得到检验结果表格"></a>18. 根据表达矩阵及样本分组信息进行批量T检验，得到检验结果表格</h2><p><a href="https://www.statmethods.net/stats/ttest.html" target="_blank" rel="noopener">t.test </a></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br></pre></td><td class="code"><pre><span class="line">rownames(exprSet) &lt;- ids[match(rownames(exprSet), ids$probe_id), <span class="number">2</span>]</span><br><span class="line"><span class="comment"># gain the matrix contains the sample names(colnames) and the gene symbol(rownames)</span></span><br><span class="line">dat = exprSet</span><br><span class="line">dim(dat)</span><br><span class="line"><span class="comment"># [1] 11460    22</span></span><br><span class="line">group_list=as.factor(group_list)</span><br><span class="line"><span class="comment"># levels provides access to the levels attribute of a variable. The first form returns the value of the levels of its argument and the second sets the attribute.</span></span><br><span class="line"><span class="comment"># levels(group_list)</span></span><br><span class="line">group1 = which(group_list == levels(group_list)[<span class="number">1</span>])</span><br><span class="line">group2 = which(group_list == levels(group_list)[<span class="number">2</span>])</span><br><span class="line">dat1 = dat[, group1]</span><br><span class="line">dat2 = dat[, group2]</span><br><span class="line">dat = cbind(dat1, dat2)</span><br><span class="line"><span class="comment"># apply() Returns a vector or array or list of values obtained by applying a function to margins of an array or matrix.</span></span><br><span class="line"><span class="comment"># 1 indicates rows, 2 indicates columns</span></span><br><span class="line">pvals = apply(exprSet, <span class="number">1</span>, <span class="keyword">function</span>(x)&#123;</span><br><span class="line">  t.test(as.numeric(x)~group_list)$p.value</span><br><span class="line">  &#125;)</span><br><span class="line">p.adj = p.adjust(pvals, method = <span class="string">"BH"</span>)</span><br><span class="line">avg_1 = rowMeans(dat1)</span><br><span class="line">avg_2 = rowMeans(dat2)</span><br><span class="line"><span class="comment"># log2(a/b) = log2(a) - log2(b)</span></span><br><span class="line">log2FC = avg_2-avg_1</span><br><span class="line">DEG_t.test = cbind(avg_1, avg_2, log2FC, pvals, p.adj)</span><br><span class="line">DEG_t.test[<span class="number">1</span>:<span class="number">5</span>,<span class="number">1</span>:<span class="number">5</span>]</span><br><span class="line"><span class="comment">#           avg_1    avg_2      log2FC     pvals     p.adj</span></span><br><span class="line">MAPK3   <span class="number">5.409914</span> <span class="number">5.538191</span>  <span class="number">0.12827699</span> <span class="number">0.5355478</span> <span class="number">0.9946078</span></span><br><span class="line">TIE1    <span class="number">2.343562</span> <span class="number">2.317535</span> -<span class="number">0.02602712</span> <span class="number">0.4370106</span> <span class="number">0.9946078</span></span><br><span class="line">CYP2C19 <span class="number">3.389422</span> <span class="number">3.424209</span>  <span class="number">0.03478680</span> <span class="number">0.6064461</span> <span class="number">0.9946078</span></span><br><span class="line">CXCR5   <span class="number">1.114136</span> <span class="number">1.147225</span>  <span class="number">0.03308823</span> <span class="number">0.5487370</span> <span class="number">0.9946078</span></span><br><span class="line">CXCR5   <span class="number">7.482762</span> <span class="number">7.362510</span> -<span class="number">0.12025210</span> <span class="number">0.6829745</span> <span class="number">0.9946078</span></span><br><span class="line"><span class="comment"># ordered by pvals</span></span><br><span class="line">DEG_t.test=DEG_t.test[order(DEG_t.test[,<span class="number">4</span>]),]</span><br><span class="line">DEG_t.test[<span class="number">1</span>:<span class="number">5</span>,<span class="number">1</span>:<span class="number">5</span>]</span><br><span class="line"><span class="comment">#           avg_1    avg_2     log2FC        pvals     p.adj</span></span><br><span class="line">SGSM2  <span class="number">7.875615</span> <span class="number">8.791753</span>  <span class="number">0.9161377</span> <span class="number">1.629755e-05</span> <span class="number">0.1867699</span></span><br><span class="line">PDE8A  <span class="number">6.622749</span> <span class="number">7.965007</span>  <span class="number">1.3422581</span> <span class="number">4.058944e-05</span> <span class="number">0.2211373</span></span><br><span class="line">DLEU1  <span class="number">7.616197</span> <span class="number">5.786041</span> -<span class="number">1.8301554</span> <span class="number">6.965416e-05</span> <span class="number">0.2211373</span></span><br><span class="line">LDOC1  <span class="number">4.456446</span> <span class="number">2.152471</span> -<span class="number">2.3039752</span> <span class="number">8.993339e-05</span> <span class="number">0.2211373</span></span><br><span class="line">USP6NL <span class="number">5.988866</span> <span class="number">7.058738</span>  <span class="number">1.0698718</span> <span class="number">9.648226e-05</span> <span class="number">0.2211373</span></span><br><span class="line">DEG_t.test=as.data.frame(DEG_t.test)</span><br><span class="line">head(DEG_t.test)</span><br><span class="line"><span class="comment">#           avg_1    avg_2     log2FC        pvals     p.adj</span></span><br><span class="line">SGSM2  <span class="number">7.875615</span> <span class="number">8.791753</span>  <span class="number">0.9161377</span> <span class="number">1.629755e-05</span> <span class="number">0.1867699</span></span><br><span class="line">PDE8A  <span class="number">6.622749</span> <span class="number">7.965007</span>  <span class="number">1.3422581</span> <span class="number">4.058944e-05</span> <span class="number">0.2211373</span></span><br><span class="line">DLEU1  <span class="number">7.616197</span> <span class="number">5.786041</span> -<span class="number">1.8301554</span> <span class="number">6.965416e-05</span> <span class="number">0.2211373</span></span><br><span class="line">LDOC1  <span class="number">4.456446</span> <span class="number">2.152471</span> -<span class="number">2.3039752</span> <span class="number">8.993339e-05</span> <span class="number">0.2211373</span></span><br><span class="line">USP6NL <span class="number">5.988866</span> <span class="number">7.058738</span>  <span class="number">1.0698718</span> <span class="number">9.648226e-05</span> <span class="number">0.2211373</span></span><br><span class="line">COMMD4 <span class="number">4.157971</span> <span class="number">3.407405</span> -<span class="number">0.7505660</span> <span class="number">2.454557e-04</span> <span class="number">0.3192169</span></span><br></pre></td></tr></table></figure>
<h2 id="19-使用limma包对表达矩阵及样本分组信息进行差异分析，得到差异分析表格，重点看logFC和P值，画个火山图-就是logFC和-log10-P值-的散点图。-。"><a href="#19-使用limma包对表达矩阵及样本分组信息进行差异分析，得到差异分析表格，重点看logFC和P值，画个火山图-就是logFC和-log10-P值-的散点图。-。" class="headerlink" title="19.使用limma包对表达矩阵及样本分组信息进行差异分析，得到差异分析表格，重点看logFC和P值，画个火山图(就是logFC和-log10(P值)的散点图。)。"></a>19.使用limma包对表达矩阵及样本分组信息进行差异分析，得到差异分析表格，重点看logFC和P值，画个火山图(就是logFC和-log10(P值)的散点图。)。</h2><p><a href="https://science.nature.nps.gov/im/datamgmt/statistics/r/formulas/" target="_blank" rel="noopener">Statistical Formulas</a></p>
<p><a href="https://vip.biotrainee.com/d/116-limma" target="_blank" rel="noopener">用limma对芯片数据做差异分析</a></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># data1: Expression matrix</span></span><br><span class="line"><span class="keyword">library</span>(limma)</span><br><span class="line"><span class="comment"># data2: creat the grouping matrix</span></span><br><span class="line"><span class="comment"># model.matrix: creates a design (or model) matrix</span></span><br><span class="line">design &lt;- model.matrix(~ -<span class="number">1</span> + factor(group_list))</span><br><span class="line">colnames(design)=levels(factor(group_list))</span><br><span class="line">rownames(design)=colnames(exprSet)</span><br><span class="line">design</span><br><span class="line"><span class="comment"># data3: Difference comparison matrix</span></span><br><span class="line"><span class="comment"># paste0: Concatenate vectors after converting to character.</span></span><br><span class="line"><span class="comment"># makeContrasts: Construct the contrast matrix corresponding to specified contrasts of a set of parameters.</span></span><br><span class="line">contrast.matrix&lt;-makeContrasts(paste0(unique(group_list),collapse = <span class="string">"-"</span>),</span><br><span class="line">                               levels = design)</span><br><span class="line">contrast.matrix</span><br><span class="line"><span class="comment"># step1</span></span><br><span class="line"><span class="comment"># lmFit Fit linear model for each gene given a series of arrays</span></span><br><span class="line">fit &lt;- lmFit(exprSet,design)</span><br><span class="line"><span class="comment"># step2</span></span><br><span class="line"><span class="comment"># Given a linear model fit to microarray data, compute estimated coefficients and standard errors for a given set of contrasts.</span></span><br><span class="line">fit2 &lt;- contrasts.fit(fit, contrast.matrix) </span><br><span class="line">fit2 &lt;- eBayes(fit2) <span class="comment">## default no trend !!!</span></span><br><span class="line"><span class="comment"># eBayes() with trend=TRUE</span></span><br><span class="line"><span class="comment"># step3</span></span><br><span class="line"><span class="comment"># topTable</span></span><br><span class="line">tempOutput = topTable(fit2, coef=<span class="number">1</span>, n=<span class="literal">Inf</span>)</span><br><span class="line">nrDEG = na.omit(tempOutput)</span><br><span class="line">write.csv(nrDEG,<span class="string">"limma_notrend.results.csv"</span>,quote = <span class="literal">F</span>)</span><br><span class="line">head(nrDEG)	</span><br><span class="line"><span class="comment">#            ID      logFC  AveExpr         t      P.Value  adj.P.Val        B</span></span><br><span class="line"><span class="number">8601</span>  TBC1D2B -<span class="number">1.0284628</span> <span class="number">5.620700</span> -<span class="number">5.835508</span> <span class="number">8.344767e-06</span> <span class="number">0.03035725</span> <span class="number">3.257356</span></span><br><span class="line"><span class="number">5522</span>    CLIC1  <span class="number">0.9888221</span> <span class="number">9.954273</span>  <span class="number">5.771215</span> <span class="number">9.672884e-06</span> <span class="number">0.03035725</span> <span class="number">3.139428</span></span><br><span class="line"><span class="number">3360</span>    DLEU1  <span class="number">1.8301554</span> <span class="number">6.950685</span>  <span class="number">5.736157</span> <span class="number">1.048607e-05</span> <span class="number">0.03035725</span> <span class="number">3.074853</span></span><br><span class="line"><span class="number">299</span>    SH3BP2 -<span class="number">1.3835699</span> <span class="number">4.463438</span> -<span class="number">5.731636</span> <span class="number">1.059590e-05</span> <span class="number">0.03035725</span> <span class="number">3.066513</span></span><br><span class="line"><span class="number">6286</span>    GPM6A  <span class="number">2.5471980</span> <span class="number">6.915045</span>  <span class="number">5.038363</span> <span class="number">5.360926e-05</span> <span class="number">0.11768171</span> <span class="number">1.752783</span></span><br><span class="line"><span class="number">10468</span>  YTHDC2 -<span class="number">0.5187135</span> <span class="number">7.602354</span> -<span class="number">4.878259</span> <span class="number">7.839394e-05</span> <span class="number">0.11768171</span> <span class="number">1.440739</span></span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">DEG=nrDEG</span><br><span class="line"><span class="comment"># abs(x) computes the absolute value of x</span></span><br><span class="line">logFC_cutoff &lt;- with(DEG,mean(abs( logFC)) + <span class="number">2</span>*sd(abs( logFC)) )</span><br><span class="line"><span class="comment"># [1] 0.5682807</span></span><br><span class="line">DEG$change = as.factor(ifelse(DEG$P.Value &lt; <span class="number">0.05</span> &amp; abs(DEG$logFC) &gt; logFC_cutoff,</span><br><span class="line">                              ifelse(DEG$logFC &gt; logFC_cutoff ,<span class="string">'UP'</span>,<span class="string">'DOWN'</span>),<span class="string">'NOT'</span>)</span><br><span class="line">)</span><br><span class="line">this_tile &lt;- paste0(<span class="string">'Cutoff for logFC is '</span>,round(logFC_cutoff,<span class="number">3</span>),</span><br><span class="line">                    <span class="string">'\nThe number of up gene is '</span>,nrow(DEG[DEG$change ==<span class="string">'UP'</span>,]) ,</span><br><span class="line">                    <span class="string">'\nThe number of down gene is '</span>,nrow(DEG[DEG$change ==<span class="string">'DOWN'</span>,])</span><br><span class="line">)</span><br><span class="line"></span><br><span class="line">g = ggplot(data=DEG, aes(x=logFC, y=-log10(P.Value), color=change)) +</span><br><span class="line">  geom_point(alpha=<span class="number">0.4</span>, size=<span class="number">1.75</span>) +</span><br><span class="line">  theme_set(theme_set(theme_bw(base_size=<span class="number">20</span>)))+</span><br><span class="line">  xlab(<span class="string">"log2 fold change"</span>) + ylab(<span class="string">"-log10 p-value"</span>) +</span><br><span class="line">  ggtitle( this_tile ) + theme(plot.title = element_text(size=<span class="number">15</span>,hjust = <span class="number">0.5</span>))+</span><br><span class="line">  scale_colour_manual(values = c(<span class="string">'blue'</span>,<span class="string">'black'</span>,<span class="string">'red'</span>)) <span class="comment">## corresponding to the levels(res$change)</span></span><br><span class="line">print(g)</span><br></pre></td></tr></table></figure>
<p><img src="https://github.com/CLDIAO/photo-for-blog-1.md/raw/master/R-20/volcano.png" alt=""></p>
<h2 id="20-对T检验结果的P值和limma包差异分析的P值画散点图，看看哪些基因相差很大。"><a href="#20-对T检验结果的P值和limma包差异分析的P值画散点图，看看哪些基因相差很大。" class="headerlink" title="20.对T检验结果的P值和limma包差异分析的P值画散点图，看看哪些基因相差很大。"></a>20.对T检验结果的P值和limma包差异分析的P值画散点图，看看哪些基因相差很大。</h2>
      
    </div>
    <footer class="article-footer">
      <a data-url="https://cldiao.github.io/2018/09/27/R语言20题/" data-id="cjv879dqp000rf0mvy0ru7hzn" class="article-share-link">Share</a>
      
<section id="comments">
<!-- 多说评论框 start -->
  <div class="ds-thread" data-thread-key="2018/09/27/R语言20题/" data-title="R语言20题-学习笔记" data-url=""></div>
<!-- 多说评论框 end -->
<!-- 多说公共JS代码 start (一个网页只需插入一次) -->
<script type="text/javascript">
var duoshuoQuery = {short_name:"chaooo"};
  (function() {
    var ds = document.createElement('script');
    ds.type = 'text/javascript';ds.async = true;
    ds.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//static.duoshuo.com/embed.js';
    ds.charset = 'UTF-8';
    (document.getElementsByTagName('head')[0] 
     || document.getElementsByTagName('body')[0]).appendChild(ds);
  })();
  </script>
<!-- 多说公共JS代码 end -->
</section>

      
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/R-表达矩阵，heatmap-PCA-T检验，火山图，ID转换/">R, 表达矩阵，heatmap, PCA, T检验，火山图，ID转换</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2018/10/18/GSEA分析/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          GSEA分析
        
      </div>
    </a>
  
  
    <a href="/2018/09/02/hello-world/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">Hello World</div>
    </a>
  
</nav>

  
</article>

</section>
        
          <aside id="sidebar">
  
    

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tag Cloud</h3>
    <div class="widget tagcloud">
      <a href="/tags/GEO/" style="font-size: 10px;">GEO</a> <a href="/tags/GSEA/" style="font-size: 10px;">GSEA</a> <a href="/tags/GSEA/" style="font-size: 10px;">GSEA,</a> <a href="/tags/LINUX-VMware-RNA-Seq-R/" style="font-size: 10px;">LINUX, VMware, RNA-Seq, R</a> <a href="/tags/Linux-shell-bash/" style="font-size: 20px;">Linux, shell, bash</a> <a href="/tags/Linux，shell-bash/" style="font-size: 10px;">Linux，shell, bash</a> <a href="/tags/Mac-office-Vmware-fusion-Ps，Adobe/" style="font-size: 10px;">Mac, office, Vmware fusion, Ps，Adobe</a> <a href="/tags/Pubmed-R-visualization/" style="font-size: 10px;">Pubmed, R, visualization</a> <a href="/tags/R-表达矩阵，heatmap-PCA-T检验，火山图，ID转换/" style="font-size: 10px;">R, 表达矩阵，heatmap, PCA, T检验，火山图，ID转换</a> <a href="/tags/R2-database-correlation/" style="font-size: 10px;">R2, database, correlation</a> <a href="/tags/RNA-Seq/" style="font-size: 10px;">RNA-Seq</a> <a href="/tags/SP-FCAS/" style="font-size: 10px;">SP, FCAS</a> <a href="/tags/TFBS/" style="font-size: 10px;">TFBS</a> <a href="/tags/blog-hexo-github/" style="font-size: 10px;">blog,hexo,github</a> <a href="/tags/linux-shell-bash/" style="font-size: 15px;">linux, shell, bash</a> <a href="/tags/lipoplex/" style="font-size: 10px;">lipoplex</a> <a href="/tags/markdown/" style="font-size: 10px;">markdown</a> <a href="/tags/scholar-IF-publications/" style="font-size: 10px;">scholar, IF, publications</a> <a href="/tags/windows10-activation/" style="font-size: 10px;">windows10, activation</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2019/04/13/Linux命令行与shell脚本编程学习第十二章/">Linux命令行与shell脚本编程学习第十二章</a>
          </li>
        
          <li>
            <a href="/2019/04/07/Linux命令行与shell脚本编程学习第十一章/">Linux命令行与shell脚本编程学习第十一章</a>
          </li>
        
          <li>
            <a href="/2019/04/06/Linux命令行与shell脚本编程学习第十章/">Linux命令行与shell脚本编程学习第十章</a>
          </li>
        
          <li>
            <a href="/2019/04/05/Linux命令行与shell脚本编程学习第九章/">Linux命令行与shell脚本编程学习第九章</a>
          </li>
        
          <li>
            <a href="/2019/04/03/Linux命令行与shell脚本编程大全学习第六章/">Linux命令行与shell脚本编程学习第六章</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2019 Chaoliang Diao<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  <link rel="stylesheet" href="/fancybox/jquery.fancybox.css">
  <script src="/fancybox/jquery.fancybox.pack.js"></script>


<script src="/js/script.js"></script>



  </div>
</body>
</html>