<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  <title>EDITING | Stay hungry, stay foolish.</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="description" content="RNA-Seq analysis in RSome preparation:Packages used:limma edgeR gplots org.Mm.eg.db RColorBrewer Glimma VMware installation:vmware-download bio-linux-download bio-linux-installation vmware-tools-insta">
<meta name="keywords" content="LINUX, VMware, RNA-Seq, R">
<meta property="og:type" content="article">
<meta property="og:title" content="EDITING">
<meta property="og:url" content="https://cldiao.github.io/RNA-Seq analysis in R/index.html">
<meta property="og:site_name" content="Stay hungry, stay foolish.">
<meta property="og:description" content="RNA-Seq analysis in RSome preparation:Packages used:limma edgeR gplots org.Mm.eg.db RColorBrewer Glimma VMware installation:vmware-download bio-linux-download bio-linux-installation vmware-tools-insta">
<meta property="og:locale" content="en">
<meta property="og:updated_time" content="2018-08-04T15:38:00.740Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="EDITING">
<meta name="twitter:description" content="RNA-Seq analysis in RSome preparation:Packages used:limma edgeR gplots org.Mm.eg.db RColorBrewer Glimma VMware installation:vmware-download bio-linux-download bio-linux-installation vmware-tools-insta">
  
    <link rel="alternate" href="/atom.xml" title="Stay hungry, stay foolish." type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    
  
  <link rel="stylesheet" href="/css/style.css">
  

</head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    
    <div id="header-inner" class="inner">
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="https://cldiao.github.io"></form>
      </div>
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      
    </div>
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">Stay hungry, stay foolish.</a>
      </h1>
      
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main"><article id="post-RNA-Seq analysis in R" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/RNA-Seq analysis in R/" class="article-date">
  <time datetime="2018-07-28T16:28:44.000Z" itemprop="datePublished">Jul 29, 2018</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      EDITING
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <!-- Table of Contents -->
        
        <h1 id="RNA-Seq-analysis-in-R"><a href="#RNA-Seq-analysis-in-R" class="headerlink" title="RNA-Seq analysis in R"></a>RNA-Seq analysis in R</h1><h2 id="Some-preparation"><a href="#Some-preparation" class="headerlink" title="Some preparation:"></a>Some preparation:</h2><h3 id="Packages-used"><a href="#Packages-used" class="headerlink" title="Packages used:"></a>Packages used:</h3><p>limma</p>
<p>edgeR</p>
<p>gplots</p>
<p>org.Mm.eg.db</p>
<p>RColorBrewer</p>
<p>Glimma</p>
<h3 id="VMware-installation"><a href="#VMware-installation" class="headerlink" title="VMware installation:"></a>VMware installation:</h3><p><a href="http://www.zdfans.com/html/5928.html" target="_blank" rel="noopener">vmware-download</a></p>
<p><a href="http://environmentalomics.org/bio-linux-download/" target="_blank" rel="noopener">bio-linux-download</a></p>
<p><a href="http://environmentalomics.org/bio-linux-installation/" target="_blank" rel="noopener">bio-linux-installation</a></p>
<p><a href="https://www.jb51.net/softjc/189144.html" target="_blank" rel="noopener">vmware-tools-installation</a></p>
<p><a href="https://www.cnblogs.com/huangjianxin/p/6343881.html" target="_blank" rel="noopener">filessystem for share</a></p>
<p><a href="https://www.netsarang.com/download/down_form.html?code=622" target="_blank" rel="noopener">X shell-download</a></p>
<p><a href="http://www.bio-info-trainee.com/tag/%E6%9C%8D%E5%8A%A1%E5%99%A8" target="_blank" rel="noopener">other common questions</a></p>
<p><a href="http://blog.sina.com.cn/s/blog_54f07aba0101s1br.html" target="_blank" rel="noopener">R语言编辑器：notepad++</a></p>
<h2 id="Overview"><a href="#Overview" class="headerlink" title="Overview"></a>Overview</h2><ul>
<li>Reading in table of counts</li>
<li>Filtering lowly expressed genes</li>
<li>Quality control</li>
<li>Normalisation for composition bias</li>
<li>Differential expression analysis</li>
<li>Testing relative to a threshold</li>
<li>Visualisation</li>
<li>Gene set testing</li>
</ul>
<h3 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h3><p>There are many steps involved in analysising an RNA-Seq experiment. Analysising an RNASeq experiment begins with sequencing reads. These are aligned to a reference genome, then the number of reads mapped to each gene can be counted. This results in a table of counts, which is what we perform statistical analyses on in R. While mapping and counting are important and necessary tasks, today we will be starting from the count data and getting stuck into analysis.</p>
<h3 id="Data-import"><a href="#Data-import" class="headerlink" title="Data import"></a>Data import</h3><p>Load all the packages will need to analysis the data.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">library(edgeR)</span><br><span class="line">library(limma)</span><br><span class="line">library(Glimma)</span><br><span class="line">library(gplots)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">library(RColorBrewer)</span><br></pre></td></tr></table></figure>
<h3 id="Mouse-mammary-gland-dataset"><a href="#Mouse-mammary-gland-dataset" class="headerlink" title="Mouse mammary gland dataset"></a>Mouse mammary gland dataset</h3><p>The data comes from a Nature Cell Biology paper, <a href="http://www.ncbi.nlm.nih.gov/pubmed/25730472" target="_blank" rel="noopener"><em>EGF-mediated induction of Mcl-1 at the switch to lactation is essential for alveolar cell survival</em></a>. Both the raw data (sequence reads) and processed data (counts) can be downloaded from <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60450" target="_blank" rel="noopener">GSE60450</a>.</p>
<p>This study examines the expression profiles of basal stem-cell enriched cells (B) and committed luminal cells (L) in the mammary gland of virgin, pregnant  and lactating mice. Six groups are present, with one for each combination of cell type and mouse status. Each group contains two biological replicates. We will first use the counts file as a starting point for out analysis. This data has already been aligned to the mouse genome.</p>
<h3 id="Reading-in-the-data"><a href="#Reading-in-the-data" class="headerlink" title="Reading in the data"></a>Reading in the data</h3><p>Download and read in the data.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"># Read the data into R</span><br><span class="line">seqdata &lt;- read.delim(&quot;GSE60450_Lactation-GenewiseCounts.txt&quot;, stringsAsFactors = FALSE)</span><br><span class="line"># Read the sample information into R</span><br><span class="line">sampleinfo &lt;- read.delim(&quot;SampleInfo.txt&quot;)</span><br><span class="line"># Take a look at the data</span><br><span class="line">head(seqdata)</span><br><span class="line"># See how many rows and columns the data frame has</span><br><span class="line">dim(seqdata)</span><br></pre></td></tr></table></figure>
<p>The <code>seqdata</code> object contains information about genes, the first column has the Entrez gene id, the second has the gene length and the remaining columns contain information about the number of reads aligning to the gene in each experimental sample. The sampleinfo file contains basic information about the samples that we will need for the analysis today.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sampleinfo</span><br></pre></td></tr></table></figure>
<p>We will manipulate and reformat the counts matrix into a suitable format for downstream analysis. The first columns in the <code>seqdata</code> dataframe contain annotation information. We need to make a new matrix containing only the counts, but we can store the gene identifiers (the <code>EntrezGeneID</code> column) as rownames. We will add more annotation information about each gene later on in the workshop.</p>
<h3 id="Format-the-data"><a href="#Format-the-data" class="headerlink" title="Format the data"></a>Format the data</h3><p>Create a new data object, <code>countdata</code>, that contains only the counts for the 12 samples.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"># Remove first two columns from seqdata</span><br><span class="line">countdata &lt;- seqdata[,-(1:2)]</span><br><span class="line"># Look at the output</span><br><span class="line">head(countdata)</span><br><span class="line"># Store EntrezGeneID as rownames</span><br><span class="line">rownames(countdata) &lt;- seqdata[,1]</span><br><span class="line"># Take a look at the output</span><br><span class="line">head(countdata)</span><br><span class="line"># Take a look at the column names</span><br><span class="line">colnames(countdata)</span><br></pre></td></tr></table></figure>
<p>The sample names are pretty long, so we’ll shorten these to contain only the relevant information about each sample. We will use the <code>substr</code> command to extract the first 7 characters and use these as the colnames.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"># using substr, you extract the characters starting at position 1 and stopping at position 7 of the colnames</span><br><span class="line">colnames(countdata) &lt;- substr(colnames(countdata),start=1,stop=7)</span><br><span class="line"># Take a look at the output</span><br><span class="line">head(countdata)</span><br></pre></td></tr></table></figure>
<p>The column names are now the same as <code>SampleName</code> in the <code>sampleinfo</code> file, which means the sample information in <code>sampleinfo</code> is in the same order as the columns in  <code>countdata</code>.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">table(colnames(countdata)==sampleinfo$SampleName)</span><br></pre></td></tr></table></figure>
<h3 id="Filtering-to-remove-lowly-expressed-genes"><a href="#Filtering-to-remove-lowly-expressed-genes" class="headerlink" title="Filtering to remove lowly expressed genes"></a>Filtering to remove lowly expressed genes</h3><p><strong>Genes with very low counts</strong> across all libraries provide little evidence for differential expression and they interfere with some of the statistical approximations that are used later. They also add to the multiple testing burden when estimating false discovery rates, reducing power to detect differentially expressed genes. These genes <strong>should be filtered out prior to further analysis</strong>.</p>
<p>There are a few ways to filter out lowly expressed genes. When there are biological replicates in each group, in this case we have a sample size of 2 in each group, we favour filtering on a minimum counts per million threshold present in at least 2 samples. Two represents the smallest sample size for each group in out experiment. In this dataset, we choose to retain genes if they are expressed at a counts-per-million (CPM) above 0.5 in at least two samples.</p>
<p>We will use the <code>cpm</code> function from the <em>edgeR</em> library to generate the CPM values and then filter.</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://cldiao.github.io/RNA-Seq analysis in R/" data-id="cjlku19810000s8mvlqm2mezv" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/LINUX-VMware-RNA-Seq-R/">LINUX, VMware, RNA-Seq, R</a></li></ul>

    </footer>
  </div>
  
    
 <script src="/jquery/jquery.min.js"></script>
  <div id="random_posts">
    <h2>Recommended Posts</h2>
    <div class="random_posts_ul">
      <script>
          var random_count =4
          var site = {BASE_URI:'/'};
          function load_random_posts(obj) {
              var arr=site.posts;
              if (!obj) return;
              // var count = $(obj).attr('data-count') || 6;
              for (var i, tmp, n = arr.length; n; i = Math.floor(Math.random() * n), tmp = arr[--n], arr[n] = arr[i], arr[i] = tmp);
              arr = arr.slice(0, random_count);
              var html = '<ul>';
            
              for(var j=0;j<arr.length;j++){
                var item=arr[j];
                html += '<li><strong>' + 
                item.date + ':&nbsp;&nbsp;<a href="' + (site.BASE_URI+item.uri) + '">' + 
                (item.title || item.uri) + '</a></strong>';
                if(item.excerpt){
                  html +='<div class="post-excerpt">'+item.excerpt+'</div>';
                }
                html +='</li>';
                
              }
              $(obj).html(html + '</ul>');
          }
          $('.random_posts_ul').each(function () {
              var c = this;
              if (!site.posts || !site.posts.length){
                  $.getJSON(site.BASE_URI + 'js/posts.js',function(json){site.posts = json;load_random_posts(c)});
              } 
               else{
                load_random_posts(c);
              }
          });
      </script>
    </div>
  </div>

    
<nav id="article-nav">
  
    <a href="/Mac上安装一些应用/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          Mac上安装一些应用
        
      </div>
    </a>
  
  
    <a href="/hello-world/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">Hello World</div>
    </a>
  
</nav>

  
</article>
 
     
  <div class="comments" id="comments">
    
     
       
      <div id="cloud-tie-wrapper" class="cloud-tie-wrapper"></div>
    
       
      
      
  </div>
 
  

</section>
           
    <aside id="sidebar">
  
    

  
    
    <div class="widget-wrap">
    
      <div class="widget" id="toc-widget-fixed">
      
        <strong class="toc-title">Content</strong>
        <div class="toc-widget-list">
              <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#RNA-Seq-analysis-in-R"><span class="toc-number">1.</span> <span class="toc-text">RNA-Seq analysis in R</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Some-preparation"><span class="toc-number">1.1.</span> <span class="toc-text">Some preparation:</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Packages-used"><span class="toc-number">1.1.1.</span> <span class="toc-text">Packages used:</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#VMware-installation"><span class="toc-number">1.1.2.</span> <span class="toc-text">VMware installation:</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Overview"><span class="toc-number">1.2.</span> <span class="toc-text">Overview</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Introduction"><span class="toc-number">1.2.1.</span> <span class="toc-text">Introduction</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Data-import"><span class="toc-number">1.2.2.</span> <span class="toc-text">Data import</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Mouse-mammary-gland-dataset"><span class="toc-number">1.2.3.</span> <span class="toc-text">Mouse mammary gland dataset</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Reading-in-the-data"><span class="toc-number">1.2.4.</span> <span class="toc-text">Reading in the data</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Format-the-data"><span class="toc-number">1.2.5.</span> <span class="toc-text">Format the data</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Filtering-to-remove-lowly-expressed-genes"><span class="toc-number">1.2.6.</span> <span class="toc-text">Filtering to remove lowly expressed genes</span></a></li></ol></li></ol></li></ol>
          </div>
      </div>
    </div>

  
    

  
    
  
    
  
    

  
    
  
    <!--微信公众号二维码-->


  
    
  
</aside>

      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-left">
      &copy; 2014 - 2018 Rumble&nbsp;|&nbsp;
      Theme by <a href="https://github.com/giscafer/hexo-theme-cafe/" target="_blank">Cafe</a>
    </div>
     <div id="footer-right">
      Contact&nbsp;|&nbsp;dior30102@gmail.com
    </div>
  </div>
</footer>
 <script src="/jquery/jquery.min.js"></script>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    <img class="back-to-top-btn" src="/images/fly-to-top.png"/>
<script>
// Elevator script included on the page, already.
window.onload = function() {
  var elevator = new Elevator({
    selector:'.back-to-top-btn',
    element: document.querySelector('.back-to-top-btn'),
    duration: 1000 // milliseconds
  });
}
</script>
      

  
    <script>
      var cloudTieConfig = {
        url: document.location.href, 
        sourceId: "",
        productKey: "e2fb4051c49842688ce669e634bc983f",
        target: "cloud-tie-wrapper"
      };
    </script>
    <script src="https://img1.ws.126.net/f2e/tie/yun/sdk/loader.js"></script>
    

  







<!-- author:forvoid begin -->
<!-- author:forvoid begin -->

<!-- author:forvoid end -->

<!-- author:forvoid end -->


  
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        tex2jax: {
          inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
          processEscapes: true,
          skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
        }
      })
    </script>

    <script type="text/x-mathjax-config">
      MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for (i=0; i < all.length; i += 1) {
          all[i].SourceElement().parentNode.className += ' has-jax';
        }
      })
    </script>
    <script type="text/javascript" src="https://cdn.rawgit.com/mathjax/MathJax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  


 <script src="/js/is.js"></script>


  <link rel="stylesheet" href="/fancybox/jquery.fancybox.css">
  <script src="/fancybox/jquery.fancybox.pack.js"></script>


<script src="/js/script.js"></script>
<script src="/js/elevator.js"></script>
  </div>
</body>
</html>